packages/mui-material/src/Popover/Popover.js

```diff
@@ -308,6 +316,20 @@ const Popover = React.forwardRef(function Popover(inProps, ref) {
     setIsPositioned(true);
   }, [getPositioningStyle]);
 
+  // Recalculate position when user scrolls
+  const [scrollTop, setScrollTop] = React.useState(0);
+  React.useEffect(() => {
+    const onScroll = () => {
+      const containerWindow = ownerWindow(resolveAnchorEl(anchorEl));
+      setScrollTop(containerWindow.scrollY);
+
+      setPositioningStyles();
+    };
+
+    window.addEventListener('scroll', onScroll);
+    return () => window.removeEventListener('scroll', onScroll);
+  }, [anchorEl, scrollTop, setPositioningStyles]);
+
```

Comment:
I think this can be simplified to:

```suggestion
  // Recalculate position when user scrolls
  React.useEffect(() => {
    window.addEventListener('scroll', setPositioningStyles);
    return () => window.removeEventListener('scroll', setPositioningStyles);
  }, [anchorEl, setPositioningStyles]);
```

Reply:
Also, we should add the event listener only if `disableScrollLock` is true